internal/ctrlflow/ctrlflow.go

```diff
@@ -0,0 +1,184 @@
+package ctrlflow
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"go/types"
+	"log"
+	mathrand "math/rand"
+	"strconv"
+	"strings"
+
+	"golang.org/x/tools/go/ast/astutil"
+	"golang.org/x/tools/go/ssa"
+	ah "mvdan.cc/garble/internal/asthelper"
+	"mvdan.cc/garble/internal/ssa2ast"
+)
+
+const (
+	mergedFileName = "GARBLE_controlflow.go"
+	directiveName  = "//garble:controlflow"
+	importPrefix   = "___garble_import"
+
+	defaultSplit  = 0
+	defaultJunk   = 0
+	defaultPasses = 1
+)
+
+type directiveParamMap map[string]string
+
+func (m directiveParamMap) GetInt(name string, def int) int {
+	rawVal, ok := m[name]
+	if !ok {
+		return def
+	}
+
+	val, err := strconv.Atoi(rawVal)
+	if err != nil {
+		panic(fmt.Errorf("invalid flag %s format: %v", name, err))
+	}
+	return val
+}
+
+// parseDirective parses a directive string and returns a map of directive parameters.
+// Each parameter should be in the form "key=value" or "key"
+func parseDirective(directive string) (directiveParamMap, bool) {
+	if !strings.HasPrefix(directive, directiveName) {
```

Comment:
a bit nicer with `strings.CutPrefix` I think, because then the directive prefix isn't part of the fields.